server/main.py

```diff
@@ -1,8 +1,119 @@
-from fastapi import FastAPI
+from typing import Annotated
+from fastapi import Depends, FastAPI, HTTPException, Query
+from sqlalchemy.orm import Session
+
+import crud, models, schemas
+from database import SessionLocal, engine
+
+models.Base.metadata.create_all(bind=engine)
 
 app = FastAPI()
 
 
+# Dependency
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
 @app.get("/")
-async def root():
-    return {"message": "Hello World"}
+def root():
+    return {"Snap Sorter": "In Development"}
+
+
+# User endpoints
+
+# Create a user
+
+
+@app.post("/users/")
+def create_user(user: schemas.UserCreate, db: Session = Depends(get_db)):
+    db_user = crud.get_user_by_bits_id(db, bits_id=user.bits_id)
+    if db_user:
+        raise HTTPException(status_code=400, detail="User already registered")
+    return crud.create_user(db=db, user=user)
+
+
+# Get a user by id
+
+
+@app.get("/users/{user_id}")
```

Comment:
Make these query parameters instead of path params:
e.g. Use `/users?id=2` or `/users?bits_id=2021...` instead of `/users/{user_id}` or `/users/bits_id/{user_id}`